.github/actions/gradle-cache/action.yml

```diff
@@ -5,4 +5,5 @@ runs:
   steps:
     - uses: gradle/gradle-build-action@v2
       with:
-        cache-read-only: ${{ github.ref != 'refs/heads/dev' }}
+        cache-write-only: ${{ github.ref_name == 'dev' }}
+        cache-read-only: ${{ startsWith(github.ref, 'refs/pull/') }}
```

Comment:
Using `github.ref != 'refs/heads/dev'` would lead to using cache on master/main branch in the future